### PR TITLE
[Feature] Enable wrap hpbandster for FedEx

### PR DESCRIPTION
**Pull request by rayrayraykk**
As the title says.

Run:
`python federatedscope/hpo.py --cfg federatedscope/example_configs/femnist/avg/bo_kde_wrap.yaml`
`python federatedscope/hpo.py --cfg federatedscope/example_configs/femnist/avg/bo_kde.yaml`
`python federatedscope/hpo.py --cfg federatedscope/example_configs/femnist/avg/bohb_wrap.yaml`
`python federatedscope/hpo.py --cfg federatedscope/example_configs/femnist/avg/bohb.yaml`

### CODE REVIEW

File: federatedscope/autotune/algos.py [hunk 1]
```diff
@@ -372,6 +372,7 @@ def _setup(self):
                         for j in range(self._cfg.hpo.table.num - 1))
             arms['arm0'] = dict(
                 (k, v) for k, v in trial_cfg.items() if k in arms['arm1'])
+            os.makedirs(self._cfg.hpo.working_folder, exist_ok=True)
```
Comment (joneswong) [inline federatedscope/autotune/algos.py:375]: I am not sure whether this is an appropriate position to do this thing.

File: federatedscope/autotune/fedex/server.py [hunk 2]
```diff
@@ -81,7 +81,8 @@ def __init__(self,
               self).__init__(ID, state, config, data, model, client_num,
                              total_round_num, device, strategy, **kwargs)
 
-        if self._cfg.federate.restore_from != '':
+        if self._cfg.federate.restore_from != '' and os.path.exists(
+                self._cfg.federate.restore_from):
```
Comment (joneswong) [inline federatedscope/autotune/fedex/server.py:85]: if it is not am empty string but no such a file, we'd better logger.warn sth.

**Review by joneswong — DISMISSED** [review]
LGTM

File: federatedscope/autotune/algos.py (hunk 1 above)
Comment (rayrayraykk) [reply federatedscope/autotune/algos.py]: Updated.

File: federatedscope/autotune/fedex/server.py (hunk 2 above)
Comment (rayrayraykk) [reply federatedscope/autotune/fedex/server.py]: Updated.

**Review by joneswong — APPROVED** [review]
approved.

**Pull request merged**